components/hab/src/main.rs

```diff
@@ -653,6 +654,49 @@ fn sub_plan_init(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
     )
 }
 
+fn sub_plan_render(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
+    let template_path = match m.value_of("TEMPLATE_PATH") {
```

Comment:
So something like this?:

```
-    let template_path = match m.value_of("TEMPLATE_PATH") {
-        Some(o) => Path::new(o),
-        None => {
-            return Err(Error::CryptoCLI("No config to render specified".to_string()));
-        }
-    };
+    let template_path = Path::new(&m.value_of("TEMPLATE_PATH").unwrap());
```

@christophermaier had mentioned somewhere else that we generally try to avoid using `unwrap()` in the "real world" (paraphrasing), and I think `match` was my attempted to "work around" `unwrap()`ing here.

Reply:
We try to avoid unwrap except in situations where a failure is logically impossible. In this case, we know that `TEMPLATE_PATH` is required to have a value, because it's configured as a required argument:
```rust
                (@arg TEMPLATE_PATH: +required {file_exists} "Path to config to render")
```
If it weren't set, clap would exit the program before we ever got to this code. It would be nice if clap could return a different type for required arguments and then we could leverage the compiler to check these things, but currently that's not available. Per [the clap docs](https://docs.rs/clap/2.32.0/clap/struct.ArgMatches.html#examples):
```rust
// The ArgMatches::value_of method returns an Option because the user may not have supplied
// that argument at runtime. But if we specified that the argument was "required" as we did
// with the "out" argument, we can safely unwrap because `clap` verifies that was actually
// used at runtime.
println!("Value for --output: {}", matches.value_of("out").unwrap());
```